src/server/pfs/server/driver.go

```diff
@@ -656,16 +653,23 @@ func (d *driver) createProjectInTransaction(txnCtx *txncontext.TransactionContex
 }
 
 func (d *driver) inspectProject(ctx context.Context, project *pfs.Project) (*pfs.ProjectInfo, error) {
-	pi := &pfs.ProjectInfo{}
-	if err := d.projects.ReadOnly(ctx).Get(pfsdb.ProjectKey(project), pi); err != nil {
-		return nil, errors.EnsureStack(err)
+	var pi *pfs.ProjectInfo
+	var err error
+	if err = dbutil.WithTx(ctx, d.env.DB, func(ctx context.Context, tx *pachsql.Tx) error {
```

Comment:
I don't love leaking the error between scopes like this.  In general `if err = ...` is the source of many mistakes, and if the linter were more flexible I'd have banned it by now.  I think in this case, moving `var error` into the callback makes the most sense.

```go
if err := dbutil.WithTx(..., func(...){ 
   var err error
   pi, err = coredb.GetProjectByName()
   ...
}
[x]
```
The way there is no `err` defined at `[x]`, which makes it less likely to be accidentally misused later.